pytrickle/stream_processor.py

```diff
@@ -68,8 +72,25 @@ async def send_data(self, data: str):
         if self.server.current_client is None:
             logger.warning("No active client connection, cannot send data")
             return False
-        await self.server.current_client.publish_data(data)
-        return True
+        
+        client = self.server.current_client
+        
+        # Check if client is in error state or stopping
+        if client.error_event.is_set() or client.stop_event.is_set():
+            logger.debug("Client is in error/stop state, not sending data")
+            return False
+        
+        # Check if protocol is shutting down
```

Comment:
Can the protocol set the client stop_event/error_event to start the tear down so we don't have to monitor for both here?